plugins/faustwp/includes/rest/callbacks.php

```diff
@@ -476,3 +488,49 @@ function handle_rest_telemetry_decision_callback( \WP_REST_Request $request ) {
 	);
 	return rest_ensure_response( $response );
 }
+
+/**
+ * Callback for WordPress register_rest_route() 'callback' parameter.
+ *
+ * Handle POST /faustwp/v1/validate_secret_key response.
+ *
+ * @link https://developer.wordpress.org/reference/functions/register_rest_route/
+ * @link https://developer.wordpress.org/rest-api/extending-the-rest-api/routes-and-endpoints/#endpoint-callback
+ *
+ * @param \WP_REST_Request $request Current \WP_REST_Request object.
+ *
+ * @return mixed A \WP_REST_Response, or \WP_Error.
+ */
+function handle_rest_validate_secret_key_callback( \WP_REST_Request $request ) {
+	return new \WP_REST_Response(
+		sprintf(
+			/* Translators: %s is replaced with the emoji indicating a successful sync. */
+			esc_html__( 'Secret key validated!', 'faustwp' ),
+			'✅'
+		),
+		200
+	);
+}
+
+/**
+ * Callback to check permissions for requests to `faustwp/v1/validate_secret_key`.
+ *
+ * Authorized if the 'secret_key' settings value and http header 'x-faustwp-secret' match.
+ *
+ * @link https://developer.wordpress.org/reference/functions/register_rest_route/
+ * @link https://developer.wordpress.org/rest-api/extending-the-rest-api/routes-and-endpoints/#permissions-callback
+ *
+ * @param \WP_REST_Request $request The current \WP_REST_Request object.
+ *
+ * @return bool True if current user can, false if else.
+ */
+function rest_validate_secret_key_permission_callback( \WP_REST_Request $request ) {
+	$secret_key = get_secret_key();
+	$header_key = $request->get_header( 'x-faustwp-secret' );
+
+	if ( $secret_key && $header_key ) {
+		return $secret_key === $header_key;
+	}
+
+	return false;
+}
```

Comment:
We already have a function that all other rest endpoints share to validate the secret key. This can be kept more dry by doing:

```suggestion
function rest_validate_secret_key_permission_callback( \WP_REST_Request $request ) {
	return rest_authorize_permission_callback( $request );
}
```